raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -108,6 +109,36 @@ def is_valid_refund(original_transfer, refund_sender, refund_transfer):
     )
 
 
+def is_valid_refund2(
+        original_transfer: LockedTransferUnsignedState,
+        refund_transfer: LockedTransferState):
+    """ True if the refund transfer matches the original transfer. """
+    refund_transfer_sender = refund_transfer.balance_proof.sender
+
+    # Ignore a refund from the target
+    if refund_transfer_sender == original_transfer.target:
+        return False
+
+    return (
+        original_transfer.identifier == refund_transfer.identifier and
+        original_transfer.lock.amount == refund_transfer.lock.amount and
+        original_transfer.lock.hashlock == refund_transfer.lock.hashlock and
+        original_transfer.target == refund_transfer.target and
+
+        # The refund transfer is not tied to the other direction of the same
+        # channel, it may reach this node through a different route depending
+        # on the path finding strategy
+        # original_receiver == refund_transfer_sender and
```

Comment:
And here `original_transfer.initiator == refund_transfer.target`

Reply:
About the same thing as above.

Reply:
btw, maybe we should uncomment this line, my rationale was for commenting it out is described here:

https://github.com/raiden-network/raiden/issues/490

But I'm not sure if the lock.amount will be equal in that case anyways.

Reply:
Lets leave it as it is for now, we can fix issues like this after your changes are merged.